### PR TITLE
Feature/dark mode support

### PR DESCRIPTION
**Pull request by drsounds**
## Hey, I just made a Pull Request!

I added support for Dark mode through the window API

````js
window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches
````
Screenshot:

CAVAT: This means that the the ui wont respond to color theme changes from system settings until page reload.
#### :heavy_check_mark: Checklist

- [ ] All tests are passing `yarn test`
- [x] Screenshots attached (for UI changes)
- [ ] Relevant documentation updated
- [ ] Prettier run on changed files
- [ ] Tests added for new functionality
- [ ] [CODEOWNERS](./CODEOWNERS) updated (for new stuff)
- [ ] Regression tests added for bug fixes

### CODE REVIEW

**Comment by drsounds** [comment]
I couldn't find the file CODEOWNERS to put my name

**Comment by Rugvip** [comment]
Thanks for the PR!

What do you think about exporting two themes from the theme file, and then switching the theme [here](https://github.com/spotify/backstage/blob/29bc05acbd12f13ed61a5ae77de49bcf5eb14744/packages/app/src/App.tsx#L57) instead? Should let us do dark mode toggle without reload, and have a switch button to configure it instead of just the media query way :grin:

**Comment by drsounds** [comment]
That might sounds interesting! I will try this out

File: packages/app/src/App.tsx [hunk 1]
```diff
@@ -32,7 +35,7 @@ const useStyles = makeStyles(theme => ({
     body: {
       height: '100%',
       fontFamily: theme.typography.fontFamily,
-      'overscroll-behavior-y': 'none',
+      'overscroll-behavior-y': 'none'
```
[the remark this answers is not in the document]
Comment (freben) [reply packages/app/src/App.tsx]: Ah yeah I see above that other formatting also breaks with the prettier standard. You may want to configure your editor to autoformat

File: packages/app/src/App.tsx [hunk 2]
```diff
@@ -48,17 +51,39 @@ const AppComponent = app.build();
...
-      <ThemeProvider theme={BackstageTheme}>
-        <ErrorDisplay forwarder={errorDialogForwarder} />
-        <Router>
-          <Root>
-            <AppComponent />
-          </Root>
-        </Router>
+    <ThemeContext.Provider value={{theme, toggleTheme}}>
+      <ThemeProvider theme={theme === 'dark' ? backstageThemeDark : backstageThemeLight}>
+        <CssBaseline>
```
Comment (freben) [inline packages/app/src/App.tsx]: Hm! Is the baseline intentionally inside the theme? I thought it needed to be the other way around

File: packages/app/src/ThemeContext.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from "react";
+export interface ThemeContextInterface {
```
Comment (freben) [inline packages/app/src/ThemeContext.tsx]: We try to stick with `type` over `interface`.

**Review by freben — COMMENTED** [review]
This is super cool!  Thank you! Couple nitpicks in here, plus you need to set your editor to use prettier and reformat the files.

File: packages/app/src/App.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (Rugvip) [reply packages/app/src/App.tsx]: Definitely need to get prettier into CI

File: packages/app/src/App.tsx (hunk 2 above)
Comment (drsounds) [reply packages/app/src/App.tsx]: Yes it is needed to be in order t o work

File: packages/app/src/App.tsx [hunk 4]
```diff
@@ -15,14 +15,16 @@
  */
 
 import { CssBaseline, makeStyles, ThemeProvider } from '@material-ui/core';
-import { BackstageTheme, createApp } from '@backstage/core';
-import React, { FC } from 'react';
+import { BackstageThemeLight, BackstageThemeDark, createApp } from '@backstage/core';
+import React, {FC} from 'react';
```
Comment (stefanalund) [inline packages/app/src/App.tsx]: Looks like Prettier hasn't been run here.

File: packages/app/src/App.tsx [hunk 5]
```diff
@@ -15,14 +15,16 @@
...
 import { BrowserRouter as Router } from 'react-router-dom';
 import Root from './components/Root';
 import ErrorDisplay from './components/ErrorDisplay';
 import * as plugins from './plugins';
 import apis, { errorDialogForwarder } from './apis';
 
+import {ThemeContext, ThemeContextType, useThemeType} from './ThemeContext';
```
Comment (stefanalund) [inline packages/app/src/App.tsx]: Same as above, no Prettier.

Comment (drsounds) [reply packages/app/src/App.tsx]: Hi, I've prettified the files now and pushed a new commit this morning, did you see it?

**Comment by drsounds** [comment]
Hi I now prettified the files could you review qgain?

**Comment by freben** [comment]
So, this actually doesn't build, it coredumps node :)

**Comment by drsounds** [comment]
Yes weird because it builds on my computer

**Comment by drsounds** [comment]
And now it was completely messed up when trying to revert the failed commit

**Comment by drsounds** [comment]
Hi, I decided to start from scratch and begin a pure new PR based on the current state of the repository

**Pull request closed without merging**